pyfolio/risk.py

```diff
@@ -0,0 +1,461 @@
+    ax.legend(loc='upper left')
+
+    return ax
+
+
+def plot_cap_exposures_gross(gross_exposures, ax=None):
+    '''
+    Plots outputs of compute_cap_exposures as area charts
+
+    Parameters
+    ----------
+    gross_exposures : array
+        Arrays of gross market cap exposures (output of compute_cap_exposures).
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    colors = ['#FF9999', '#FFCC99', '#99FF99', '#99CCFF', '#CC99FF']
+
+    ax.stackplot(gross_exposures[0].index, gross_exposures,
+                 labels=CAP_NAMES, colors=colors, baseline='zero')
+    ax.axhline(0, color='k', linestyle='-')
+    ax.set_title('Market Cap Exposures: Gross')
+    ax.set_xlabel('Date')
+    ax.set_ylabel('Proportion of Gross Exposure \n in Market Cap Buckets')
+    ax.legend(loc='upper left')
+
+    return ax
+
+
+def plot_cap_exposures_net(net_exposures, ax=None):
+    '''
+    Plots outputs of compute_cap_exposures as line graphs
+
+    Parameters
+    ----------
+    net_exposures : array
+        Arrays of gross market cap exposures (output of compute_cap_exposures).
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    colors = ['#FF9999', '#FFCC99', '#99FF99', '#99CCFF', '#CC99FF']
+
+    for i in range(len(net_exposures)):
+        ax.plot(net_exposures[i], color=colors[i], label=CAP_NAMES[i])
+    ax.axhline(0, color='k', linestyle='-')
+    ax.set_title('Market Cap Exposures: Net')
+    ax.set_xlabel('Date')
+    ax.set_ylabel('Proportion of Net Exposure \n in Market Cap Buckets')
+    ax.legend(loc='upper left')
+
+    return ax
+
+
+def compute_volume_exposures(shares_held, volumes, percentile):
+    '''
+    Returns arrays of pth percentile of long, short and gross volume exposures
+    of an algorithm's held shares
+
+    Parameters
+    ----------
+    shares_held : pd.DataFrame
+        Daily number of shares held by an algorithm.
+        - See full explanation in create_risk_tear_sheet
+
+    volume : pd.DataFrame
+        Daily volume per asset
+        - See full explanation in create_risk_tear_sheet
+
+    percentile : float
+        Percentile to use when computing and plotting volume exposures
+        - See full explanation in create_risk_tear_sheet
+    '''
+
+    shares_held = shares_held.replace(0, np.nan)
+
+    shares_longed = shares_held[shares_held > 0]
+    shares_shorted = -1 * shares_held[shares_held < 0]
+    shares_grossed = shares_held.abs()
+
+    longed_frac = shares_longed.divide(volumes)
+    shorted_frac = shares_shorted.divide(volumes)
+    grossed_frac = shares_grossed.divide(volumes)
+
+    longed_threshold = 100*longed_frac.quantile(percentile, axis=1)
+    shorted_threshold = 100*shorted_frac.quantile(percentile, axis=1)
+    grossed_threshold = 100*grossed_frac.quantile(percentile, axis=1)
+
+    return longed_threshold, shorted_threshold, grossed_threshold
+
+
+def plot_volume_exposures_longshort(longed_threshold, shorted_threshold,
```

Comment:
Specified by Jonathan: we take the number of shares held at the end of the day, divided by the total volume. That gives us a number per stock per day. We find the 10th percentile (i.e. the most illiquid) and plot that as a time series. So it is a measure of how exposed our portfolio is to illiquid stocks.